x/emissions/keeper/keeper.go

```diff
@@ -3996,17 +3997,44 @@ func (k *Keeper) PruneReputerNonces(ctx context.Context, topicId uint64, blockHe
 }
 
 // Return true if the nonce is within the worker submission window for the topic
-// Inclusive of the start block height and exclusive of the end block height
-func (k *Keeper) BlockWithinWorkerSubmissionWindowOfNonce(topic types.Topic, nonce types.Nonce, blockHeight int64) bool {
-	return nonce.BlockHeight <= blockHeight && blockHeight < topic.WorkerSubmissionWindow+nonce.BlockHeight
+// Inclusive of the start block height and of the end block height.
+func (k *Keeper) BlockWithinWorkerSubmissionWindowOfNonce(topic types.Topic, nonce types.Nonce, blockHeight int64) (bool, error) {
```

Comment:
```suggestion
func BlockWithinWorkerSubmissionWindowOfNonce(topic types.Topic, nonce types.Nonce, blockHeight int64) (bool, error) {
```

We may not attach this func to the `Keeper` if we don't need to access its internal elements

Reply:
Good point. I think I want to change it to another file though.

Reply:
done